packages/serverpod/lib/src/database/concepts/table.dart

```diff
@@ -5,13 +5,11 @@ import 'package:serverpod_serialization/serverpod_serialization.dart';
 /// Holds data corresponding to a row in the database. Concrete classes are
 /// typically generated. Instances of [TableRow] can also be serialized and
 /// either passed to clients or cached.
-abstract class TableRow implements SerializableModel {
-  /// Create a new TableRow object.
-  TableRow([this.id]);
-
+abstract interface class TableRow implements SerializableModel {
   /// The id column of the row. Can be null if this row is not yet stored in
   /// the database.
-  int? id;
+  int? get id;
+  set id(int? value);
```

Comment:
Question: Do we really need to change this to a setter and a getter?

It is essentially the same thing but seems a bit bloated.

Seems like it is fine to define a field directly in an interface that needs to be
 overridden by anything implementing the interface:
 
 ```dart
abstract interface class MyInterface {
  int? myInt;
}

class MyClass implements MyInterface {
  @override
  int? myInt;
}

void interfaceMethod<T extends MyInterface>(T myInterface) {
  print(myInterface.myInt);
}

void main() {
  MyClass myClass = MyClass();
  myClass.myInt = 5;
  interfaceMethod(myClass);
}
 ```

This prints out "5".

Reply:
Then we can also remove the code generated constructor logic for setting a private `_id` field.